lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -155,6 +155,44 @@ struct ConstantOpConv : public OpConversionPattern<ConstantOp> {
   }
 };
 
+struct NamedConstantOpConv : public OpConversionPattern<NamedConstantOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(NamedConstantOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Type resultType = typeConverter->convertType(op.getResult().getType());
+    auto value = op.getValue();
+    std::string symStr;
+    switch (op.getKind()) {
+    case NamedConst::Parameter:
+      symStr = "parameter";
+      break;
+    case NamedConst::LocalParameter:
+      symStr = "localparameter";
+      break;
+    case NamedConst::SpecParameter:
+      symStr = "specparameter";
+      break;
+    }
+    symStr += ":";
+    symStr += op.getNameAttr().str();
+    auto symAttr = rewriter.getStringAttr(symStr);
+    if (isa<IntegerAttr>(value)) {
+      auto valueInt = cast<IntegerAttr>(value);
+      auto paramOp =
+          rewriter.create<hw::ParamValueOp>(op->getLoc(), resultType, valueInt);
```

Comment:
I would make this an `hw.constant`, since you're basically using the `ParamValueOp` to materialize a constant attribute.